CMakeLists.txt

```diff
@@ -13,6 +13,13 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
                            -Werror=float-conversion)
 endif()
 
+if (NOT WIN32)
+  # Make all executables position independent on Linux
+  set(CMAKE_EXE_LINKER_FLAGS -pie)
+  list(APPEND CMAKE_C_FLAGS -fPIC)
+  list(APPEND CMAKE_CXX_FLAGS -fPIC)
+endif()
+
```

Comment:
I believe this is not necessary.
You can the same fPIC-option to our conanfile and then conan will take care of the rest.

This way the user has full control and can disable fPIC on the package manager level if necessary and it automatically propagates to all deps.

Reply:
On the other hand by forcing -pie we kind of making -fPIC mandatory, if we disable it in conanfile.py our project will stop compiling.

Reply:
Ah I see. In this case I would still add the fPIC option and add a check for its value in the configure function:
```
if self.settings.os != "Windows" and not self.options.fPIC:
  raise ConanInvalidConfiguration("We only support compiling with fPIC enabled!")
```

That has the advantage that trying to compile with fPIC-disabled does not silently fail.

Reply:
done